lib/hiera/backend.rb

```diff
@@ -233,6 +249,16 @@ def lookup(key, default, scope, order_override, resolution_type)
       def clear!
         @backends = {}
       end
+
+      def qualified_lookup(segments, hash)
+        value = hash
+        segments.each do |segment|
+          break if value.nil? || value == :undefined
```

Comment:
:undefined can be removed - it is from a time when it was used in Puppet. The tests for :undefined should be removed too.